src/mixins/portfolio.js

```diff
@@ -43,45 +41,58 @@ export default {
     return {
       wallet: undefined,
       userInfo: null,
-      collectedNFTs: [],
-      sellingNFTClassIds: [],
       currentTab: ['collected', 'created'].includes(this.$route.query.tab)
         ? this.$route.query.tab
         : 'created',
-      isLoading: true,
+      isLoading: false,
     };
   },
   computed: {
-    ...mapGetters(['getNFTClassIdSorter']),
+    ...mapGetters(['getNFTClassIdSorter', 'getNFTClassIdListByAddress']),
     userAvatar() {
       return this.userInfo?.avatar;
     },
     userDisplayName() {
       return this.userInfo?.displayName || this.wallet;
     },
-    collectedNFTClassIds() {
-      const classIdSet = new Set(this.collectedNFTs.map(n => n.classId));
-      return Array.from(classIdSet);
+    nftClassIds() {
+      return this.getNFTClassIdListByAddress(this.wallet);
     },
-    sortedCollectedNFTClassIds() {
-      return this.getNFTClassIdSorter(this.collectedNFTClassIds);
+
+    // for userStats
+    collectedClassIds() {
+      return this.nftClassIds?.collected || [];
+    },
+    createdClassIds() {
+      return this.nftClassIds?.created || [];
+    },
+
+    // for NFTCardItems
+    sortedCollectedClassIds() {
+      const collectedClassIds = [...new Set(this.collectedClassIds)];
+      return this.getNFTClassIdSorter(
+        (collectedClassIds?.length && collectedClassIds) || []
+      );
     },
-    sortedSellingNFTClassIds() {
-      return this.getNFTClassIdSorter(this.sellingNFTClassIds);
+    sortedCreatedClassIds() {
+      return this.getNFTClassIdSorter(
+        (this.createdClassIds?.length && this.createdClassIds) || []
+      );
```

Comment:
Seem no need to guard as you have already guarded in `this.createdClassIds`
```suggestion
      return this.getNFTClassIdSorter(this.createdClassIds);
```